core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/CachingIterableTest.java

```diff
@@ -51,8 +50,9 @@ public void shouldCacheSmallIterable(@Mock final Iterable<Integer> mockIterable)
 
         final Iterable<Integer> cachingIterable = new CachingIterable<>(mockIterable, 5);
 
-        assertThat(SMALL_LIST).isEqualTo(Lists.newArrayList(cachingIterable));
-        assertThat(SMALL_LIST).isEqualTo(Lists.newArrayList(cachingIterable));
+        assertThat(cachingIterable)
+                .containsExactlyElementsOf(SMALL_LIST)
+                .containsExactlyElementsOf(SMALL_LIST);
```

Comment:
I'm not sure why this was duplicated previously, but I can't see a reason for it to remain. Same for the `LARGE_LIST ` assertion below.

Reply:
I think it is to test that the iterable doesn't get exhausted after being evaluated

Reply:
Ok, in that case how sure are we sure that these assertions can test for that?